### PR TITLE
feat: no more backend features

### PR DESCRIPTION
**Pull request by haraldh**
also remove the `exec-wasmtime-bin` crate again.

Fixes: https://github.com/enarx/enarx/issues/1798

### CODE REVIEW

File: tests/exec/mod.rs [hunk 1]
```diff
@@ -75,7 +75,7 @@ fn unix_echo() {
     handle.join().unwrap();
 }
 
-#[cfg(feature = "backend-sev")]
+#[cfg(use_shim)]
```
Comment (rvolosatovs) [inline tests/exec/mod.rs:78]: this should already be applied in the `integration.rs`

**Review by rvolosatovs — COMMENTED** [review]
yes, please! :partying_face:

File: tests/exec/mod.rs (hunk 1 above)
Comment (haraldh) [reply tests/exec/mod.rs]: removed

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -25,19 +25,16 @@ jobs:
...
-            flags: --no-default-features --features=backend-sgx
           - name: kvm
             path: .
             host: [ self-hosted, linux ]
-            flags: --no-default-features --features=backend-kvm
           - name: nil
             path: .
             host: [ self-hosted, linux ]
-            flags: --no-default-features --features=backend-nil --test integration --bin enarx
+            flags: "--test integration --bin enarx -- wasm::"
```
Comment (rvolosatovs) [inline .github/workflows/coverage.yml:37]: Does this really work?

Comment (haraldh) [reply .github/workflows/coverage.yml:37]: yes... see https://github.com/enarx/enarx/runs/6350302648?check_suite_focus=true

**Comment by bstrie** [comment]
How does this relate to https://github.com/enarx/enarx/pull/1826 ? Are we closing that PR in favor of this one?

**Comment by haraldh** [comment]
> How does this relate to #1826 ? Are we closing that PR in favor of this one?

@bstrie

It is just a Draft based upon some state of #1826 rebased on master...

The last patch of this PR will be rebased on master once #1826 is merged

File: Cargo.toml [hunk 3]
```diff
@@ -31,39 +26,42 @@ bitflags = { version = "1.2", default-features = false }
...
+sallyport = { version = "0.4.0", path = "crates/sallyport", default-features = false }
+semver = { version = "1.0", default-features = false }
+sgx = { version = "0.5.0", features = ["rcrypto"], default-features = false }
+static_assertions = { version = "1.1.0", default-features = false }
+ureq = { version = "2.4.0", default-features = false }
+vdso = { version = "0.2", default-features = false }
+x86_64 = { version = "0.14.9", default-features = false }
+
+# binary dependencies
 # FIXME: The enarx-exec-wasmtime-bin crate should be merged with enarx-exec-wasmtime again, see https://github.com/enarx/enarx/issues/1798
```
[the remark this answers is not in the document]
Comment (haraldh) [reply Cargo.toml:63]: @bstrie well spotted! thanks!

[the remark this answers is not in the document]
Comment (haraldh) [reply Cargo.toml]: removed

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -59,35 +60,6 @@ jobs:
             path: crates/sallyport/Cargo.toml
             target: --target=x86_64-unknown-none
 
-  clippy-single-backends:
```
Comment (rvolosatovs) [inline .github/workflows/lint.yml:62]: Why remove the clippy job?
I would expect that we'd just use `cargo clippy --workspace`

File: tests/syscall/mod.rs [hunk 5]
```diff
@@ -137,7 +137,7 @@ fn get_att() {
     run_test(bin, 0, None, None, None);
 }
 
-#[cfg(feature = "backend-sgx")]
+#[cfg(enarx_with_shim)]
```
Comment (rvolosatovs) [inline tests/syscall/mod.rs:140]: Already guarded in `test/integration.rs`

Comment (haraldh) [reply tests/syscall/mod.rs]: removed

File: tests/syscall/mod.rs [hunk 6]
```diff
@@ -137,11 +137,15 @@ fn get_att() {
     run_test(bin, 0, None, None, None);
 }
 
-#[cfg(feature = "backend-sgx")]
 #[cfg_attr(any(not(host_can_test_sgx), not(host_can_test_attestation)), ignore)]
 #[test]
 #[serial]
 fn sgx_get_att_quote() {
+    if let Ok(backend) = std::env::var("ENARX_BACKEND") {
```
Comment (rvolosatovs) [inline tests/syscall/mod.rs:144]: Is this env var documented anywhere? Is it only used for testing?

Comment (haraldh) [reply tests/syscall/mod.rs:144]: this env var is one of the official ways to set the backend and the only way for tests to do it

**Review by rvolosatovs — APPROVED** [review]

**Review by bstrie — APPROVED** [review]

**Pull request merged**